models/aether_lat-lon/model_mod.f90

```diff
@@ -0,0 +1,616 @@
+!   which calls nf90_enddef before it.
+call nc_add_global_creation_time(ncid, routine)
+
+call nc_add_global_attribute(ncid, "model_source", source, routine)
+call nc_add_global_attribute(ncid, "model", "aether", routine)
+
+call nc_end_define_mode(ncid)
+
+! Flush the buffer and leave netCDF file open
+call nc_synchronize_file(ncid)
+
+end subroutine nc_write_model_atts
+
+!-----------------------------------------------------------------------
+! Read dimension information from the template file and use 
+! it to assign values to variables.
+
+subroutine assign_dimensions()
+
+integer  :: ncid
+character(len=24), parameter :: routine = 'assign_dimensions'
+
+call error_handler(E_MSG, routine, 'reading filter input ['//trim(template_file)//']')
+
+ncid = nc_open_file_readonly(template_file, routine)
+
+! levels
+nlev = nc_get_dimension_size(ncid, trim(LEV_DIM_NAME), routine)
+allocate(levs(nlev))
+call nc_get_variable(ncid, trim(LEV_VAR_NAME), levs, routine)
+
+! latitiude
+nlat = nc_get_dimension_size(ncid, trim(LAT_DIM_NAME), routine)
+allocate(lats(nlat))
+call nc_get_variable(ncid, trim(LAT_VAR_NAME), lats, routine)
+
+! longitude
+nlon = nc_get_dimension_size(ncid, trim(LON_DIM_NAME), routine)
+allocate(lons(nlon))
+call nc_get_variable(ncid, trim(LON_VAR_NAME), lons, routine)
+
+end subroutine assign_dimensions
+
+!-----------------------------------------------------------------------
+! Parse the table of variables' characteristics into arrays for easier access.
+
+function assign_var(variables, MAX_STATE_VARIABLES) result(var)
+
+   character(len=vtablenamelength), intent(in) :: variables(:, :)
+   integer, intent(in)                         :: MAX_STATE_VARIABLES
+
+   type(var_type) :: var
+
+   integer        :: ivar
+
+   !-----------------------------------------------------------------------
+   ! Codes for interpreting the NUM_STATE_TABLE_COLUMNS of the variables table
+   integer, parameter :: NAME_INDEX      = 1 ! ... variable name
+   integer, parameter :: QTY_INDEX       = 2 ! ... DART kind
+   integer, parameter :: MIN_VAL_INDEX   = 3 ! ... minimum value if any
+   integer, parameter :: MAX_VAL_INDEX   = 4 ! ... maximum value if any
+   integer, parameter :: UPDATE_INDEX    = 5 ! ... update (state) or not
+
+   ! Loop through the variables array to get the actual count of the number of variables
+   do ivar = 1, MAX_STATE_VARIABLES
+      ! If the element is an empty string, the loop has exceeded the extent of the variables
+      if (variables(1, ivar) == '') then
+          var%count = ivar-1
+          exit
+      endif 
+   enddo
+  
+   ! Allocate the arrays in the var derived type
+   allocate(var%names(var%count), var%qtys(var%count), var%clamp_values(var%count, 2), var%updates(var%count))
+  
+   do ivar = 1, var%count
+   
+      var%names(ivar) = trim(variables(NAME_INDEX, ivar))
+  
+      var%qtys(ivar) = get_index_for_quantity(variables(QTY_INDEX, ivar))
```

Comment:
DART QTYs are case sensitive,  so the following will error:
```
model_nml
   variables      = 'Temperature',    'qty_TEMPERATURE',      'NA',  'NA',   'UPDATE',
```


to_upper will convert your character TO_UPPER

Reply:
Added subroutine call to_upper(table_entry) before the var%qtys element is assigned.